example/accel/CMakeLists.txt

```diff
@@ -9,6 +9,13 @@ project(CeleritasAccelExample VERSION 0.0.1 LANGUAGES CXX)
 cmake_policy(VERSION 3.12...3.22)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/..")
+
+include(CheckLanguage)
+check_language(CUDA)
+if(CMAKE_CUDA_COMPILER)
+  enable_language(CUDA)
+endif()
+
```

Comment:
Yes it's not pretty. In that case, we need to support both CUDA and non-CUDA environments for our CI, so we must check if NVCC is found. In simple cases `enable_language(CUDA)` with a failure should be enough. 

I suppose we can do that for the user if we're in the top-level project directory. I'm not sure how `CMAKE_PROJECT_SOURCE_DIR` and `enable_language(CUDA)` would interact when there are multiple calls to `project`, I'd guess it's fine since `project` can enable a language.